subprojects/plugins/src/main/java/org/gradle/api/tasks/Upload.java

```diff
@@ -16,143 +16,33 @@
 
 package org.gradle.api.tasks;
 
-import groovy.lang.Closure;
-import groovy.lang.DelegatesTo;
-import org.gradle.api.Action;
+import org.gradle.api.InvalidUserCodeException;
 import org.gradle.api.artifacts.Configuration;
-import org.gradle.api.artifacts.PublishException;
-import org.gradle.api.artifacts.dsl.RepositoryHandler;
-import org.gradle.api.file.FileCollection;
 import org.gradle.api.internal.ConventionTask;
-import org.gradle.api.internal.artifacts.ArtifactPublicationServices;
-import org.gradle.api.internal.artifacts.ArtifactPublisher;
-import org.gradle.api.internal.artifacts.Module;
-import org.gradle.api.internal.artifacts.configurations.ConfigurationInternal;
-import org.gradle.api.internal.artifacts.repositories.PublicationAwareRepository;
-import org.gradle.internal.Transformers;
-import org.gradle.internal.deprecation.DeprecationLogger;
-import org.gradle.util.internal.ConfigureUtil;
+import org.gradle.api.internal.DocumentationRegistry;
 import org.gradle.work.DisableCachingByDefault;
 
-import javax.annotation.Nullable;
 import javax.inject.Inject;
-import java.io.File;
-import java.util.List;
-
-import static org.gradle.util.internal.CollectionUtils.collect;
 
 /**
  * Uploads the artifacts of a {@link Configuration} to a set of repositories.
```

Comment:
We also need to update this to say that this task is no longer supported and does nothing. The type is kept around for backwards compatibility.